installers/MacInstaller.hs

```diff
@@ -55,24 +60,40 @@ travisJobIdFromEnv :: IO (Maybe String)
 travisJobIdFromEnv = lookupEnv "TRAVIS_JOB_ID"
 
 installerConfigFromEnv :: IO InstallerConfig
-installerConfigFromEnv = mkEnv <$> envAPI <*> envVersion
+installerConfigFromEnv = mkEnv <$> envAPI <*> envVersion <*> envCluster
   where
     envAPI = fromMaybe "cardano" <$> lookupEnv "API"
     envVersion = fromMaybe "dev" <$> lookupEnv "DAEDALUS_VERSION"
-    mkEnv "cardano" ver = InstallerConfig
+    envCluster = fromMaybe "mainnet" <$> lookupEnv "DAEDALUS_CLUSTER"
+    mkEnv "cardano" ver clusterName = InstallerConfig
       { icApi = "cardano"
       , predownloadChain = False
       , appNameLowercase = "daedalus"
       , appName = "Daedalus"
-      , pkg = "dist/Daedalus-installer-" <> T.pack ver <> ".pkg"
+      , pkg = "dist/Daedalus-installer-" <> T.pack ver <> "-" <> T.pack clusterName <> ".pkg"
       , appRoot = "../release/darwin-x64/Daedalus-darwin-x64/Daedalus.app"
+      , clusterName = clusterName
       }
 
 main :: IO ()
 main = do
   hSetBuffering stdout NoBuffering
 
   cfg <- installerConfigFromEnv
+
+  let capitalize :: String -> String
+      capitalize [] = []
+      capitalize (x:xs) = [Data.Char.toUpper x] <> xs
+      cluster'    = fromMaybe (error "Unrecognised cluster name in DAEDALUS_CLUSTER: should be one of:  mainnet staging") $ read $ capitalize $ clusterName cfg
```

Comment:
Why not just `read clusterName :: Cluster`

Reply:
Because `read` fails obscurely, and because the input strings are lowercase.

Reply:
Not sure I follow - you're using read anyway? I'm surprised this compiles since `fromMaybe :: a -> Maybe a -> a` and you're passing `read` output to `Maybe a`

Reply:
it's been fixed to `readMaybe` in the new commits -- and refactored into a common location : -)